client/src/main/java/com/cloudera/nav/sdk/client/writer/MetadataWriter.java

```diff
@@ -125,7 +125,7 @@ private void getMRelations(Entity entity, MClassWrapper graph) {
         entity.getClass());
     for (MRelationEntry relEntry : relationAttrs) {
       for(Entity other : relEntry.getConnectedEntities(entity)) {
-        if (!(other instanceof EndPointProxy)) {
+        if (!other.getIsEndPoint()) {
```

Comment:
Ok. Lets look at this from a user perspective. So lets take our three use cases.
1. User wants to Pre Register an HdfsEntity.
2. User wants to add tags/ custom metadata to an existing HdfsEntity.
3. User wants to link to a already existing HdfsEntity. 

In all these cases user should be just able to directly use the HdfsEntity object that we have in our models. I guess it not make a good design if we had HdfsEndPointEntity and HdfsEntity as separate classes to deal with this. Also if we just make HdfsEntity extend from an EndPointProxy then it restricts that HdfsEntity to be used for the first 2 use cases.

Lets say we have an EndPointProxy class, which can just have the Map of idAttributes. Now when the user is linking to a PigOperation, the user creates an EndPointProxy object. But how does the user know what are the attributes to be passed in to this map of idAttributes? It would have been much easier for the user to just create a PigOperation and set the end point proxy flag. The validate method of PigOperation would validate that is has all the entity attributes required to populate the id for the relation or the entity id. Let me know your thoughts on this.

Reply:
1. Custom Lineage example
Since entity x is set as an end point, this hdfs entity will be treated as the end point of the relation. Also since x is not set an end point, it is not written in the elements json.

2. Since a is not set as an End Point it will be written to the elements json.

Reply:
Sorry, I wasn't 100% clear. I meant to type Entity X in both cases (a == x). Can you see how utterly confusing it's going to get for users in these very common situations?

Reply:
ok. in CustomMetadata it is clear that we are not creating any relations and hence the user would not set the endPoint flag.In CustomLineage the user knows the HdfsEntity is going to be part of a relation and hence the user sets the end point. We just need to document what setting the flag means and how to use it which should make it clear to the user. Its similar to the previous version where the user had to create an EndPointProxy in case of a CustomLineage and create an HDfsEntity in case of CustomMetadata.

Reply:
It's not similar at all because now you're using Entity in both cases so you wouldn't expect to have to create separate instances for the same thing when using it in different places. The errors this is going to create will be utterly impossible to debug. Not sure how else I can explain it to you the evidence is abundantly clear. Don't use this pattern.